torch/_inductor/graph.py

```diff
@@ -200,7 +202,10 @@ def constant_name(self, name: str, device_override: torch.device):
         return alt_name
 
     def placeholder(self, target, args, kwargs):
-        example: torch.Tensor = super().placeholder(target, args, kwargs)
+        example: Union[torch.Tensor, SymInt] = super().placeholder(target, args, kwargs)
+        if isinstance(example, SymInt):
+            self.graph_inputs[target] = example.node.expr
+            return example.node.expr
```

Comment:
There may not be enough bindings for sizes if we only extract them out of tensors, because a saved activation can have a size like `s0 * 2`, and this is how you end up with dependence on s0, but NO tensor actually has a bare s0 as its input that we can easily extract.